src/mscorlib/src/System/ArraySegment.cs

```diff
@@ -160,6 +154,24 @@ public bool Equals(ArraySegment<T> obj)
             return !(a == b);
         }
 
+        private static void ThrowConstructorValidationFailedExceptions(T[] array, int offset, int count)
```

Comment:
Could take an `object` and move to ThrowHelper to reduce generic load? Throw functions could also probably be shared with `Span<T>`?